### ISSUE TITLE
Detective Credits on Traitor confirm Option

### ISSUE THREAD

**Issue by MrXonte** [TTT-2/TTT2#1693]
## Is your feature request related to a problem? Please describe.
From a gameplay perspective it feels weird to be able to "confirm" a dead body indirectly as a detective/shop inno role by getting credits once a traitor dies. This is even worse when a traitor accidentally dies or gets killed by a different team.

## Describe the solution you'd like
There should be an option to give out credits only after the body has been confirmed.

## Describe alternatives you've considered
not sure if there is an in between possible here as its mainly an information game.

**Comment by Histalek** [comment]
Fair point indeed. i'm inclined to classify this as a bug because it allows a core gameplay element to be ignored

Having the detective be awarded the credit only upon confirmation should be the only way this works. I don't think we should make this optional

**Comment by Satton2** [comment]
> Fair point indeed. i'm inclined to classify this as a bug because it allows a core gameplay element to be ignored
> 
> Having the detective be awarded the credit only upon confirmation should be the only way this works. I don't think we should make this optional

Hello!

Just in case you will change that one day:
1. There are cases where the Detective can be unable to reach the body to confirm or search. Like it was, for example, burnt by some weapons or fallen in some map entity that removes bodies. Some server and player can be not okay with that behavior, since it actually very common for bodies to be lost. I suggest making it an option, rather than making it a default thing.
2. If you make it an option, I suggest not making it a thing upon confirmation, but rather:
2.1 do not give credit reward upon a kill made by a different team in the first place.
2.2 make an option to delay giving that credit reward, so it's not actually known for sure was that killed man a traitor or not for some time (_let players think and actually check was he a traitor/some other baddie or not_). This will keep that mystery and untrustful moments well enough.
2.3 make an option to keep it the same way like it is now.
3. Add an option to "hide" the notification about credits being on a body. This is kinda similar, since not every role has a shop and credits and this may give some unwanted hints. (_I believe it was asked a while ago already_).

**Comment by MrXonte** [comment]
> 2. If you make it an option, I suggest not making it a thing upon confirmation, but rather:
>    2.1 do not give credit reward upon a kill made by a different team in the first place.
>    2.2 make an option to delay giving that credit reward, so it's not actually known for sure was that killed man a traitor or not for some time (_let players think and actually check was he a traitor/some other baddie or not_). This will keep that mystery and untrustful moments well enough.
>    2.3 make an option to keep it the same way like it is now.

2. Maybe there should be 3 options, then? Confirmation for team credit reward still seems like the ideal option. If you lose the traitor's body, you won't have gotten his regular credits either.
2.1 would be a cool option to have
2.2 sounds a bit confusing because it doesn't solve the root issue anyway, a timer delays it but in the end you still get information for free after the timer expires?

**Comment by Satton2** [comment]
>    2.2 sounds a bit confusing because it doesn't solve the root issue anyway, a timer delays it but in the end you still get information for free after the timer expires?

Do you really expect that after killing someone and someone else saw that, they will, for example, wait 30 seconds for that notification that can probably be for somebody else? :D 

> Maybe there should be 3 options, then? Confirmation for team credit reward still seems like the ideal option. If you lose the traitor's body, you won't have gotten his regular credits either.

Maybe. Or you can use timer to set it to extremely long time that never ends. xD I am joking on this one, of course, but that works tho.

**Comment by MrXonte** [comment]
> Do you really expect that after killing someone and someone else saw that, they will, for example, wait 30 seconds for that notification that can probably be for somebody else? :D
if the traitor dies from other causes, you still get a notification that tells you a traitor died, even if its with some delay that's powerful information.
If you implement the teammate killed traitor only, you also run into another problem I just noticed, that if someone kills and confirms a T, and then no credits show up, they are essentially known to be T or other Team.
And I think these scenarios do happen often enough, at least when I'm playing.
So for me the confirm is still the most viable option, although a timer option would be nice for servers that want to keep it as is (timer=0) or with an option to not lose those credits and get them after a time

> Maybe. Or you can use timer to set it to extremely long time that never ends. xD I am joking on this one, of course, but that works tho.
i mean its fair enough it would work xD but if you make the timer length variable you might as well make a bool if a timer should exist in the first place xD

**Comment by Histalek** [comment]
> 1. There are cases where the Detective can be unable to reach the body to confirm or search. Like it was, for example, burnt by some weapons or fallen in some map entity that removes bodies. Some server and player can be not okay with that behavior, since it actually very common for bodies to be lost. I suggest making it an option, rather than making it a default thing.

Agreed that playing against an exposed Traitor and them jumping down a cliff to deny their credits doesn't sound fun. But then again that's already possible today at least for unspent credits.

Regarding the burned corpse that sounds like a valid gameplay mechanic to me. A traitor burning their teammates corpse to deny other shopping roles the credits is definitely something i would want to keep being an option

We definitely should keep in mind that bodies might be lost to various reasons, although i don't have an idea on how to combat that as of now

> 2. If you make it an option, I suggest not making it a thing upon confirmation, but rather:
>   2.1 do not give credit reward upon a kill made by a different team in the first place.

This would have the potential of being an innocent tester :/

> 2.2 make an option to delay giving that credit reward, so it's not actually known for sure was that killed man a traitor or not for some time (_let players think and actually check was he a traitor/some other baddie or not_). This will keep that mystery and untrustful moments well enough.

There is no way that we will get this right for each and every round setup and general server configuration. And making this timeframe itself configurable is another complexity in the server settings which i don't want to introduce anymore at this point

> 2.3 make an option to keep it the same way like it is now.

Yeah i don't want to face any backlash so the option will probably be there :D although it might not be the default

> 3. Add an option to "hide" the notification about credits being on a body. This is kinda similar, since not every role has a shop and credits and this may give some unwanted hints. (_I believe it was asked a while ago already_).

Yep there is definitely need for improvement here. I personally don't like credits showing when simply looking at a corpse, searching the corpse should be the least you would need to do to see credits
Additionally taking credits from a body should be a conscious choice and not coupled to confirming a body etc. etc.
But that goes a bit overboard on what was asked for in the original post
